### PR TITLE
Deeper zoom for image viewer

### PR DESCRIPTION
**Pull request by hajdam**
- Zoom +8 to -8 steps
- Fullscreen action for all viewers / moved to Viewer menu
- Added zoom to actual size / to fit view
- Added view scrolling by left mouse button holding and dragging
- Zoom by mouse click while holding Ctrl/Meta
- Zoom by mouse wheel scrolling

### CODE REVIEW

File: mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java [hunk 1]
```diff
@@ -110,86 +160,121 @@ public void setPresenter(ViewerPresenter presenter) {
...
-        InputStream in = file.getInputStream();
-        while ((read=in.read(buffer, 0, buffer.length))!=-1)
-            bout.write(buffer, 0, read);
-
-        byte imageBytes[] = bout.toByteArray();
-        bout.close();
+        InputStream in;
+        byte[] imageBytes;
+        try (ByteArrayOutputStream bout = new ByteArrayOutputStream()) {
+            in = file.getInputStream();
```
Comment (ahadas) [inline mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java:175]: how about adding `in` to the try-with-resources statement above
```java
try (ByteArrayOutputStream bout = new ByteArrayOutputStream();
     InputStream in =  file.getInputStream()) {
  ....
```

File: mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java [hunk 2]
```diff
@@ -213,70 +298,184 @@ public void open(AbstractFile file) throws IOException {
...
+        private Color backgroundColor;
+
+        ImageViewerImpl() {
+            backgroundColor = ThemeManager.getCurrentColor(Theme.EDITOR_BACKGROUND_COLOR);
             ThemeManager.addCurrentThemeListener(this);
-    	}
-    	
-    	////////////////////////
+
+            MouseAdapter ma = new MouseAdapter() {
```
Comment (ahadas) [inline mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java:379]: how about renaming that to `mouseAdapter`?

**Review by ahadas — COMMENTED** [review]
@hajdam thanks for this contribution, it looks good overall

File: mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java (hunk 1 above)
Comment (hajdam) [reply mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java]: Replaced by StreamUtis.copyStream

File: mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerSnapshot.java [hunk 3]
```diff
@@ -47,18 +47,13 @@ public final class TextViewerSnapshot implements MuSnapshotable {
      * Default line numbers value.
      */
     public static final boolean DEFAULT_LINE_NUMBERS = true;
-    /**
-     * Last known file presenter full screen mode.
-     */
-    public static final String TEXT_FILE_PRESENTER_FULL_SCREEN = TEXT_FILE_PRESENTER_SECTION + "." + "full_screen";
```
[the remark this answers is not in the document]
Comment (hajdam) [reply mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerSnapshot.java]: Plan was to add general configuration for viewers to remember window position and full screen flag for all of them and not just text viewer. I didn't get to it as I was trying to support pinch to zoom. I'll try it with next set of changes, I guess...

File: mucommander-viewer-api/build.gradle [hunk 4]
```diff
@@ -4,6 +4,8 @@ dependencies {
     api project(':mucommander-commons-file')
     api project(':mucommander-commons-util')
 
+    compileOnly 'com.google.code.findbugs:jsr305:1.3.9'
```
[the remark this answers is not in the document]
Comment (ahadas) [reply mucommander-viewer-api/build.gradle:7]: ok

**Review by ahadas — APPROVED** [review]
thanks!

**Comment by ahadas** [comment]
Version 1.2.0 that includes a fix for this issue [has been released](https://github.com/mucommander/mucommander/releases/tag/1.2.0-1)

**Pull request merged**